ra/1.1/ra-metadata-example.xml

```diff
@@ -1,103 +1,191 @@
+     type should have the same set of these parameters. This is a hint to RMs
+     and other tools, which may, but are not required to, enforce it.
+-->
+
+<!-- These attributes indicate that each instance of this agent must specify a
+     value for the "config-file" parameter, and (since no other parameter uses
+     the same unique group) the value should be unique across all instances.
+-->
+<parameter name="config-file" unique-group="config-file" required="1">
+
+<!-- A helpful, extended description of the parameter. A user interface might
+     display it to the user when asked for help with an option. You may include
+     examples, etc.
+
+     Each parameter must have at least one longdesc, and may have multiple with
+     different "lang" attributes.
+-->
+<longdesc lang="en">
+The full path to the example daemon configuration file.
+</longdesc>
+
+<!-- A brief description of the parameter's purpose. A user interface might
+     display it to the user as a tooltip or similar.
+
+     Each parameter must have at least one shortdesc, and may have multiple with
+     different "lang" attributes.
+-->
+<shortdesc lang="en">Configuration filename</shortdesc>
+
+<!-- Allowed values, and optionally the default value. Type may be boolean,
+     string, integer, or select; with select, this must have "option" child
+     elements specifying each allowed value.
+-->
+<content type="string"/>
+
+</parameter>
+
+<!-- Since the "ip" and "port" parameters share the same unique group, the
+     combination of the two should be unique across all instances of this
+     resource type.
+
+     If required is 0 or unspecified, an instance of this resource type may
+     omit this parameter (typically, a default value will be used in that
+     case).
 -->
+<parameter name="ip" unique-group="address">
 <longdesc lang="en">
-The resource name is the directory where the filesystem will be actually
-mounted.  Please make sure it exists. 
+The local IP address the example daemon should bind to and accept requests on.
+If omitted, the wildcard address will be used.
 </longdesc>
+<shortdesc lang="en">IP address</shortdesc>
+<content type="string" default="*"/>
+</parameter>
 
-<!-- The shortdesc may be displayed by the resource manager as a tooltip or
-     equivalent -->
-<shortdesc lang="en">Mountpoint</shortdesc>
+<parameter name="port" unique-group="address">
+<longdesc lang="en">
+The port number the example daemon should accept requests on.
+</longdesc>
+<shortdesc lang="en">Port number</shortdesc>
+<content type="string" default="65535"/>
+</parameter>
 
-<!-- Further definition of the content -->
-<content type="string" default="/mnt" />
+<!-- Example of a parameter that is reloadable and uses select. Reloadable
+     implies that "mode" can be changed without requiring a restart of
+     example-daemon; perhaps the resource agent writes the mode to a file
+     that example-daemon polls periodically.
```

Comment:
Thanks for the review! You're right, that sentence should be reworded like:

Reloadable implies that a change in "mode" can be made effective by calling the agent's "reload-params" action, without requiring a restart of example-daemon. As an example, perhaps the "reload-params" action contacts the daemon via a REST API to change its mode.

(I figured the REST API example is a little more relatable than a daemon that polls a file ...)

Reply:
> As an example, perhaps the "reload-params" action contacts the daemon via a REST API to change its mode.

Based on your comment https://github.com/ClusterLabs/OCF-spec/pull/24#discussion_r595538291 the RA is not supposed to bother the example-daemon by any mean...

Do I miss something?

Reply:
The RM can call reload-agent instead of stopping and starting the daemon; the agent can implement reload-agent to interact with the daemon or not as desired. The idea is that reload-agent is "cheaper" than stop+start and requires no downtime.

To fabricate another example, say a database agent has a parameter for default character locale. When the user changes that parameter in the configuration, reload-agent can maybe use the database client to change the current default, without requiring potentially large downtime from restarting the database.

Reply:
Mh, that explanation stick with `reload`, that was advertised like an action cheaper than stop+start and with no downtime.

So I have the feeling that old `reload` has been renamed `reload-agent` and a new `reload` action appears to graceful reload the resource, but from outside of the control of the RM.

Reply:
Good point, they both save downtime versus a stop+start -- the difference is whether the service's own configuration changed (reload), or its cluster configuration changed (reload-agent).

What happened to the old reload depends on which of the two incompatible interpretations you used before. :) That's why it was separated into two commands: ocf:heartbeat agents used "reload" to reload the service, while Pacemaker used it to reload the agent (only the ocf:pacemaker dummy agents supported that sense, as far as I know). Since nearly all existing agents used it to reload the service, that was kept as the main "reload" action, and Pacemaker's interpretation became "reload-agent".

Reply:
To shed more light on how we got here: OCF agents were originally designed as a sort of extension to init scripts. The idea was they could be used as either an init script or a cluster agent. Of course, it later turned out that no one ever used OCF agents as init scripts, and init scripts have gone nearly extinct anyway (in favor of systemd and a few other alternatives). But, because init scripts often supported a "reload" action to reload the service's configuration, ocf:heartbeat agents wound up adopting that convention, even though Pacemaker went a different direction.

Reply:
That history around ocf/init script help to understand.

Moreover, I never noticed the difference between Pacemaker and heartbeat providers around `reload` action!

Reply:
Changed with new "Rename reload-params action" commit